src/arche/report.py

```diff
@@ -1,102 +1,66 @@
-        if level == Level.ERROR:
-            cls.write_color_text(msg)
-        elif level == Level.WARNING:
-            cls.write_color_text(msg, color=Fore.YELLOW)
-        elif outcome == Outcome.PASSED.name:
-            cls.write_color_text(msg, color=Fore.GREEN)
+    def _rule_outcome(self, rule) -> str:
+        """
+        Returns the outcome status of a given rule
+        """
+        if not rule.messages:
+            return "Passed"
+        message_level = rule.messages.keys()
+        if any([level.value == Level.ERROR.value for level in message_level]):
+            return "Failed"
+        elif any([level.value == Level.WARNING.value for level in message_level]):
+            return "Warning"
         else:
-            cls.write(msg)
-
-    def write_details(self, short: bool = False, keys_limit: int = 10) -> None:
-        for result in self.results.values():
-            if result.detailed_messages_count:
-                display_markdown(
-                    f"{result.name} ({result.detailed_messages_count} message(s)):"
-                )
-                self.write_rule_details(result, short, keys_limit)
-            for f in result.figures:
-                f.show()
-            display_markdown("<br>")
-
-    @classmethod
-    def write_rule_details(
-        cls, result: Result, short: bool = False, keys_limit: int = 10
-    ) -> None:
-        for rule_msgs in result.messages.values():
-            for rule_msg in rule_msgs:
-                if rule_msg.errors:
-                    cls.write_detailed_errors(rule_msg.errors, short, keys_limit)
-                elif rule_msg.detailed:
-                    cls.write(rule_msg.detailed)
-
-    @classmethod
-    def write_detailed_errors(cls, errors: Dict, short: bool, keys_limit: int) -> None:
-        error_messages = sorted(errors.items(), key=lambda i: len(i[1]), reverse=True)
-
-        if short:
-            keys_limit = 5
-            error_messages = error_messages[:5]
-
-        for attribute, keys in error_messages:
-            if isinstance(keys, list):
-                keys = pd.Series(keys)
-            if isinstance(keys, set):
-                keys = pd.Series(list(keys))
-
-            sample = Report.sample_keys(keys, keys_limit)
-            display_markdown(
-                f"{len(keys)} items affected - {attribute}: {sample}", raw=True
-            )
+            return "Skipped"
+
+    def _order_rules(self, rules):
+        """
+        Returns an ordered list of Results
+        """
+        RULE_ORDER = ["Passed", "Failed", "Warning", "Skipped"]
+        rules = sorted([(RULE_ORDER.index(self._rule_outcome(rule)), rule) for rule in rules],
+                       key=lambda x: x[0]
+                       )
+        return [rule[1] for rule in rules]
+
+    def display(self) -> None:
+        for f in self.results.values():
+            f.figures
+
+        template = self.env.get_template('template.html')
+        resultHTML = template.render(
```

Comment:
`rendered` seems better for me

// TODO async output https://jinja.palletsprojects.com/en/2.10.x/api/#jinja2.Template.render_async. I guess we need to make all rules async first. (just a thought, not necessary for this pr)